packages/cli/src/commonlib/sharepointLogin.ts

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+"use strict";
+
+import { SharepointTokenProvider, UserError } from "@microsoft/teamsfx-api";
+import { LogLevel } from "@azure/msal-node";
+import { CodeFlowLogin } from "./codeFlowLogin";
+import CLILogProvider from "./log";
+import { CryptoCachePlugin } from "./cacheAccess";
+import { signedIn, signedOut } from "./common/constant";
+import { login, LoginStatus } from "./common/login";
+import axios from "axios";
+
+const accountName = "appStudio";
+
+const graphScopes = ["https://graph.microsoft.com/User.ReadBasic.All"];
+const SERVER_PORT = 0;
+
+const graphCachePlugin = new CryptoCachePlugin(accountName);
+const cachePlugin = new CryptoCachePlugin(accountName);
+
+export class SharepointLogin extends login implements SharepointTokenProvider {
+  private static instance: SharepointLogin;
+
+  private static codeFlowInstance: CodeFlowLogin;
+  private graphCodeFlowInstance: CodeFlowLogin;
```

Comment:
If getToken in ```CodeFlowLogin``` can take scopes as an parameter, do we need second codeFlowLogin instance?